webapp/src/js/services/database-connection-monitor.js

```diff
@@ -0,0 +1,28 @@
+/*
+PouchDB intermittently emits an unhandledrejection to ineffectively communicate that it has catastrophically failed.
+This service detects the event and resolves a promise when it occurs.
+
+DOMException: Failed to execute 'transaction' on 'IDBDatabase': The database connection is closing
+*/
+angular.module('inboxServices').factory('DatabaseConnectionMonitor',
+  function($window) {
+    'use strict';
+    'ngInject';
+
+    const promiseToCloseDatabase = new Promise(function(resolve) { 
+      $window.addEventListener('unhandledrejection', function(promiseRejectionEvent) {
+        if (
+          promiseRejectionEvent &&
+          promiseRejectionEvent.reason &&
+          promiseRejectionEvent.reason.message === 'Failed to execute \'transaction\' on \'IDBDatabase\': The database connection is closing.'
+        ) {
+          resolve(promiseRejectionEvent);
+        }
+      });
+    });
+
+    return {
+      onDatabaseClosed: () => promiseToCloseDatabase,
```

Comment:
I think this should be an event emitter or a callback not a promise - Promises shouldn't be resolved multiple times and it's possible this one will be.

Reply:
Okay done